Update Platform.IO manifest

mcspr opened the pull request:
**Note:** not sure where to point this: master or develop?

Update library dependencies.

Tested with
```
$ pio lib -g install file://.
$ PLATFORMIO_CI_SRC=examples/NTPClientESP8266/NTPClientESP8266.ino pio ci --lib . --lib examples/NTPClientESP8266/ -b d1_mini
```

mcspr committed:
Add Platform.IO dependencies

maxgerhardt commented:
@gmag11 Since your `library.json` requests a non-existant `Time` library version, this makes trouble for PlatformIO, i.e., it crashes.  You state

https://github.com/gmag11/NtpClient/blob/3c22f2f1e57f57e8e3b6e44656e81d9ad34bc8d8/library.json#L8-L17

but in `Time`, the latest version is `1.5`

https://github.com/PaulStoffregen/Time/blob/b05d455c91e5f0551a91af030a58f0061374ae31/library.json#L16-L20

mcspr committed:
"espressif" platform for ESPAsyncUDP

https://github.com/me-no-dev/ESPAsyncUDP/blob/b592ac6e93e9dbfd61db70ce80dcd04a2afcc76a/library.json#L16

mcspr commented:
@maxgerhardt this PR was opened before that change. i've added example on how to test with pio for exactly this reason :/

updated for master.
also, fixed espasyncudp problem. because library was not updated in 3 years, it still references old "espressif" platform. it will try to build that lib when using esp32 and fail, so we need to reference "invalid" platform for it to work properly.

*i'd still strongly suggest to avoid depending on async library -> (#60 last comment)*
